packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -855,3 +860,67 @@ contract OPContractsManager_AddGameType_Test is Test {
         );
     }
 }
+
+/// @title OPContractsManager_NoSuperchainOrProtocolVersionsUpgrade_Test
+/// @notice Tests that upgrading worldchain fails when using OPCM V2.0.0,
+///         and then passes when using the modified OPCM.
+contract OPContractsManager_NoSuperchainOrProtocolVersionsUpgrade_Test is OPContractsManager_Upgrade_Harness {
+    IOPContractsManager opcmV200;
+    address worldchainProxyAdmin;
+
+    function setUp() public override {
+        super.setUp();
+
+        // Set the fork block number to 21983965
+        vm.createSelectFork("https://eth.llamarpc.com", 21983965);
+
+        worldchainProxyAdmin = 0xA4fB12D15Eb85dc9284a7df0AdBC8B696EdbbF1d;
+
+        vm.etch(worldchainProxyAdmin, vm.getDeployedCode("test/mocks/Callers.sol:DelegateCaller"));
+
+        opcmV200 = IOPContractsManager(0x026b2F158255Beac46c1E7c6b8BbF29A4b6A7B76);
+
+        opChainConfigs[0] = IOPContractsManager.OpChainConfig({
+            systemConfigProxy: ISystemConfig(0x6ab0777fD0e609CE58F939a7F70Fe41F5Aa6300A),
+            proxyAdmin: IProxyAdmin(0xd7405BE7f3e63b094Af6C7C23D5eE33Fd82F872D),
+            absolutePrestate: absolutePrestate
+        });
+    }
+
+    /// @notice Tests that upgrading worldchain fails when using OPCM V2.0.0.
+    function test_upgrade_withV200OPCM_fails() public {
+        vm.expectRevert(bytes("Ownable: caller is not the owner"));
+        DelegateCaller(worldchainProxyAdmin).dcForward(
+            address(opcmV200), abi.encodeCall(IOPContractsManager.upgrade, (opChainConfigs))
+        );
+    }
+
+    /// @notice Tests that upgrading worldchain passes when using the modified OPCM with no superchain config or
+    ///         protocol versions upgrade
+    function test_upgrade_withModifiedOPCM_succeeds() public {
+        IOPContractsManager modifiedOPCM = opcm = IOPContractsManager(
```

Comment:
Wow, I have never seen anyone do this before. 

Assuming it actually compiles, is there a reason we need to set both `opcm ` and `modifiedOpcm` here?